src/main/java/com/microsoft/sqlserver/jdbc/ConfigurableRetryLogic.java

```diff
@@ -0,0 +1,343 @@
+    }
+
+    String getLastQuery() {
+        return lastQuery;
+    }
+
+    private static void setUpRules() throws SQLServerException {
+        stmtRules = new HashMap<>();
+        lastQuery = "";
+        LinkedList<String> temp;
+
+        if (!rulesFromConnectionString.isEmpty()) {
+            temp = new LinkedList<>();
+            Collections.addAll(temp, rulesFromConnectionString.split(";"));
+            rulesFromConnectionString = "";
+        } else {
+            temp = readFromFile();
+        }
+        createRules(temp);
+    }
+
+    private static void createRules(LinkedList<String> listOfRules) throws SQLServerException {
+        stmtRules = new HashMap<>();
+
+        for (String potentialRule : listOfRules) {
+            ConfigRetryRule rule = new ConfigRetryRule(potentialRule);
+
+            if (rule.getError().contains(",")) {
+                String[] arr = rule.getError().split(",");
+
+                for (String retryError : arr) {
+                    ConfigRetryRule splitRule = new ConfigRetryRule(retryError, rule);
+                    stmtRules.put(Integer.parseInt(splitRule.getError()), splitRule);
+                }
+            } else {
+                stmtRules.put(Integer.parseInt(rule.getError()), rule);
+            }
+        }
+    }
+
+    private static String getCurrentClassPath() {
+        try {
+            String className = new Object() {}.getClass().getEnclosingClass().getName();
+            String location = Class.forName(className).getProtectionDomain().getCodeSource().getLocation().getPath();
+            location = location.substring(0, location.length() - 16);
+            URI uri = new URI(location + "/");
+            return uri.getPath();
+        } catch (Exception e) {
+            if (CONFIGURABLE_RETRY_LOGGER.isLoggable(java.util.logging.Level.FINEST)) {
+                CONFIGURABLE_RETRY_LOGGER.finest("Unable to get current class path for properties file reading.");
+            }
+        }
+        return null;
+    }
+
+    private static LinkedList<String> readFromFile() {
+        String filePath = getCurrentClassPath();
+        LinkedList<String> list = new LinkedList<>();
+
+        try {
+            File f = new File(filePath + DEFAULT_PROPS_FILE);
+            timeLastModified = f.lastModified();
+            try (BufferedReader buffer = new BufferedReader(new FileReader(f))) {
+                String readLine;
+                while ((readLine = buffer.readLine()) != null) {
+                    if (readLine.startsWith("retryExec")) {
+                        String value = readLine.split("=")[1];
+                        Collections.addAll(list, value.split(";"));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            if (CONFIGURABLE_RETRY_LOGGER.isLoggable(java.util.logging.Level.FINEST)) {
+                CONFIGURABLE_RETRY_LOGGER.finest("No properties file exists or file is badly formatted.");

```

Comment:
Do we want to check if the file exists? eg. If file exists, parse it, and if it's formatted badly error out completely. Otherwise, if file doesn't exist continue on silently.

Right now, if we just look for the IOException we can't tell if it's a bad file.

Reply:
why are we eating the exception and not throwing?